File: 013_LeetCode_1/2_leapYear.cpp
#include<iostream>
using namespace std;

int main()
{
    int num;
    cout<<"Enter number: ";
    cin>>num;

    if(num%4==0)
    {
        if(num%100==0)
        {
            if(num%400==0)
            {
                cout<<num<<" is a Leap Year.";
            }
            else
            cout<<num<<" is a not a Leap Year.";
        }
        else
            cout<<num<<" is a Leap Year.";
    }
    else
    cout<<num<<" is not a Leap Year.";  
}



/* class Solution{
    public:
    int isLeap(int N){
        // code here

        if(N%400==0)
        return 1;
        else if(N%4==0 && N%100!=0)
        return 1;
        else
        return 0;
    }
}; */

File: 013_LeetCode_1/2_leapYear.py

n = int(input("Enter Num: "))

# Method 1
if((n%4==0 and n%100!=0) or (n%400==0)):
    print("true")
else:
    print("false")



# Method 2
print((n % 4 == 0 and n % 100 != 0) or (n % 400 == 0))



# Method 3
class Solution:
    def checkYear(self, n):
        if((n%4==0 and n%100!=0) or (n%400==0)):
            return True
        else:
            return False
sol = Solution()
print(sol.checkYear(n))



# Method 4
class Solution:
    def checkYear(self, n):
        return (n % 4 == 0 and n % 100 != 0) or (n % 400 == 0)
sol = Solution()
print(sol.checkYear(n))